packages/arb-token-bridge-ui/src/util/rpc/getRpcUrl.ts

```diff
@@ -0,0 +1,36 @@
+import { ChainId } from '../../types/ChainId'
+
+import { getInfuraRpcUrl } from './infura'
+import { getAlchemyRpcUrl } from './alchemy'
+
+type RpcProvider = 'infura' | 'alchemy'
+
+export type ProductionChainId = Exclude<
+  ChainId,
+  ChainId.Local | ChainId.ArbitrumLocal | ChainId.L3Local
+>
+
+function getRpcProvider(): RpcProvider {
+  const rpcProviderFromEnv = process.env.NEXT_PUBLIC_RPC_PROVIDER
+  return (rpcProviderFromEnv?.toLowerCase() as RpcProvider) || 'infura' // the || is intentional to handle empty strings
+}
+
+export function getRpcUrl(
+  chainId: ProductionChainId,
+  rpcProvider: RpcProvider = getRpcProvider(),
+  rpcProviderKey?: string
+): string | undefined {
```

Comment:
doesn't look like there would be a case where the return value is `undefined` since `rpcProvider` will be `infura` by default?